code.test.js

```diff
@@ -21,4 +21,4 @@ assert(JSON.stringify(augmentingPath(graph, 'foo', 'bar')) == '[]');
 
 var graph = {'foo': {},
     'bar': {'foo': 1}};
-assert(JSON.stringify(augmentingPath(graph, 'foo', 'foo')) == JSON.stringify(['foo']));
+assert(JSON.stringify(augmentingPath(graph, 'foo', 'foo')) == JSON.stringify([]));//I remove == JSON.stringify(['foo'])) because there is no path from foo to itself.
```

Comment:
...but you're already at `foo`, so this should work.

Reply:
But the 'foo' edges do not have any capacities.  I mean 
var graph = {'foo': { **'bar': 1**},
    'bar': {'foo': 1}}; it will work.

Reply:
If you want to get to `foo` and you're already at `foo`, you're done.